src/libcommon/log/sentry/handler.cpp

```diff
@@ -434,7 +446,158 @@ SentryHandler::~SentryHandler() {
     }
 }
 
-SentryHandler::SentryEvent::SentryEvent(const std::string &title, const std::string &message, SentryLevel level,
+Handler::SentryEvent::SentryEvent(const std::string &title, const std::string &message, Level level,
                                         SentryConfidentialityLevel confidentialityLevel, const SentryUser &user) :
-    title(title), message(message), level(level), confidentialityLevel(confidentialityLevel), userId(user.userId()) {}
-} // namespace KDC
+    title(title),
+    message(message), level(level), confidentialityLevel(confidentialityLevel), userId(user.userId()) {}
+
+void Handler::stopPTrace(const pTraceId &id, bool aborted) {
+    std::scoped_lock lock(_mutex);
+    if (id == 0) return;
+    auto performanceTraceIt = _performanceTraces.find(id);
+    if (performanceTraceIt == _performanceTraces.end()) {
+        return;
+    }
+    const PerformanceTrace &performanceTrace = performanceTraceIt->second;
+
+    // Stop any child PerformanceTrace
+    std::vector<pTraceId> toDelete;
+    for (const auto &childPerformanceTraceId: performanceTrace.children()) {
+        toDelete.push_back(childPerformanceTraceId); // We can't delete directly the childPerformanceTraceId from the map as we
+                                                     // are iterating over it
+    }
+    for (auto childPerformanceTraceId2: toDelete) {
+        stopPTrace(childPerformanceTraceId2, aborted);
+    }
+
+    // Stop the PerformanceTrace
+    if (performanceTrace.isSpan()) {
+        if (!performanceTrace.span()) {
+            assert(false && "Span is not valid");
+            return;
+        }
+        if (aborted) sentry_span_set_status(performanceTrace.span(), SENTRY_SPAN_STATUS_ABORTED);
+        sentry_span_finish(performanceTrace.span()); // Automatically stop any child transaction
+    } else {
+        if (!performanceTrace.transaction()) {
+            assert(false && "Transaction is not valid");
+            return;
+        }
+        if (aborted)
+            sentry_transaction_set_status(performanceTrace.transaction(),
+                                          SENTRY_SPAN_STATUS_ABORTED /*This enum is also valid for a transactrion.*/);
+        sentry_transaction_finish(performanceTrace.transaction()); // Automatically stop any child transaction
+    }
+    _performanceTraces.erase(id);
+}
+
+pTraceId Handler::startTransaction(const std::string &name, const std::string &description) {
+    sentry_transaction_context_t *tx_ctx = sentry_transaction_context_new(name.c_str(), description.c_str());
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx, sentry_value_new_null());
+    _pTraceIdCounter++;
+    pTraceId traceId = _pTraceIdCounter;
+    auto [it, res] = _performanceTraces.try_emplace(traceId, traceId);
```

Comment:
This looks strange: the key and the value coincide.

Reply:
The second argument of `try_emplace` is not the value itself but the argument that will be passed to the constructor of `PerformanceTrace`.

```cpp
namespace Sentry {
class Handler {
     ...
    private:
    std::map<pTraceId, PerformanceTrace> _performanceTraces;
    ....
}
```